src/GafferUI/AnimationGadget.cpp

```diff
@@ -1325,17 +1320,28 @@ std::string AnimationGadget::undoMergeGroup() const
 	return boost::str( boost::format( "AnimationGadget%1%%2%" ) % this % m_mergeGroupId );
 }
 
-bool AnimationGadget::onTimeAxis( const Imath::V2i& rasterPosition ) const
+bool AnimationGadget::onTimeAxis( const IECore::LineSegment3f& line ) const
 {
-	const ViewportGadget *viewportGadget = ancestor<ViewportGadget>();
-	Imath::V2i resolution = viewportGadget->getViewport();
+	Imath::V3f i;
+	if( ! line.intersect( Imath::Plane3f( Imath::V3f( 0, 0, 1 ), 0 ), i ) )
+	{
+		return false;
+	}
 
-	return rasterPosition.y >= resolution.y - m_yMargin;
+	const ViewportGadget* const viewportGadget = ancestor<ViewportGadget>();
+	return viewportGadget->worldToRasterSpace( i ).y >= ( viewportGadget->getViewport().y - m_yMargin );
```

Comment:
I know this mimics the original code, but it should be `gadgetToRasterSpace()` rather than `worldToRasterSpace()`  (at the moment the AnimationGadget has an identity transform so it doesn't matter, but the general case is that any gadget can be given a transform).

Reply:
done